src/app/shared/data-classes-list/data-classes-list.component.ts

```diff
@@ -112,7 +113,11 @@ export class DataClassesListComponent implements AfterViewInit {
           );
         }),
         map((data: any) => {
-          this.totalDataClassCount = data.body.count;
+          if (this.parentDataClass.extendsDataClasses) {
```

Comment:
I've spotted that there is legacy code in the `mdm-data-classes-list` component, where `any` is used too often. I think, seeing as you have edited this component last, this would be a good opportunity to remove `any` and replace with defined types for better compiler checks. For instance:

1. `parentDataModel` should probably be a `DataModelDetail`
2. `parentDataClass` should probably be a `DataClassDetail`
3. `isEditable` should be a `boolean`
4. And so on...

Also:

1. `loadingData` doesn't actually seem to be used in this component. If that is true, please remove it.